### PR TITLE
correct DMA region add/remove returns

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/dma.c [hunk 1]
```diff
@@ -346,16 +351,15 @@ MOCK_DEFINE(dma_controller_add_region)(dma_controller_t *dma,
...
         }
     }
 
+    ret = idx;
     dma->nregions++;
 
-    return idx;
-
-err:
-    return -idx - 1;
```
Comment (tmakatos) [inline lib/dma.c:358]: The patch looks good, however are we changing the semantics of `dma_controller_add_region` when the region can't be mapped?

Comment (jlevon) [reply lib/dma.c:358]: in what way? if dma_map_region() fails, it returns -errno, and that's what we return

Comment (tmakatos) [reply lib/dma.c:358]: I'm talking about this:
```
 * - On failure, a negative integer (-x - 1) where x is the region number
 *   where this region would have been mapped to if the call could succeed
 *   (e.g. due to conflict with existing region).
 ```

Comment (jlevon) [reply lib/dma.c:358]: oh, yes, I need to remove that comment, thanks.

**Review by tmakatos — APPROVED** [review]

**Pull request merged**